### PR TITLE
Update Prolific USB to Serial driver to 1.6.1_20170620

### PR DESCRIPTION
**Pull request by wastrachan**
Changes:
- Update Prolific USB to Serial driver to version 1.6.1_20170620
- Use official Prolific website as download source (prolificusa.com is a distributor and child company of Tectona Electronics Inc)
- Respect version limitations of MacOS Mountain Lion and below
- Add caveats stanza for kext installation

Notes:
- Version string in download URL and package file do not match in this release. 

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, if **adding a new cask**:

- [ ] Named the cask according to the [token reference].
- [ ] `brew cask install {{cask_file}}` worked successfully.
- [ ] `brew cask uninstall {{cask_file}}` worked successfully.
- [ ] Checked there are no [open pull requests] for the same cask.
- [ ] Checked the cask was not already refused in [closed issues].
- [ ] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-drivers/pulls
[closed issues]: https://github.com/caskroom/homebrew-drivers/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

**Comment by vitorgalvao** [comment]
Some things that would typically need to be addressed:

1. `url` is repeated. It can still be set only once, outside the conditionals.
2. Is there anywhere on the prolificusa.com that proves it’s part of prolific.com.tw? As a maintainer unfamiliar with this software, I’d need that to approve the change.

We appreciate the contribution, but are going to refuse it for the following reasons:

1. We officially only support the current macOS version and the one before, so it’s not a priority to support Mountain Lion and under. That is not usually an issue, but it’s also not reason enough to aprove considering the other reasons.
2. We’re losing a bit of maintainability due to the wrongly named `pkg` you pointed out.
3. And the most important, we’re losing `https` support with this change, and that’s not worth it for the rest of the changes.

All that said, if you submit another PR with the `kext` `caveat`, we can accept that.

Thank you again for the contribution!

**Comment by wastrachan** [comment]
I've leave this closed, but a couple of responses (because I hate to leave things hanging):

1. Technically the URL is different, in one of the downloads it contains a `v` before the version string. I suppose this could be handled with a base URL variable and some string interpolation.
2. It's sort of the opposite problem, actually. prolific.com.tw is the _actual_ website of prolific, the company that makes the PL2303 USB-Serial converter. prolificusa.com is a distributor and subsidiary of another company that sells the PL2303 for prolific. http://prolificusa.com/about/ states that "ProlificUSA.com was created primarily to facilitate and target the distribution of the Prolific (Technology Inc.) product line to the electronics marketplace in the Americas." The footer of the website that they are owned by Tectona Electronics, inc. That's about all we have to go on-- they're _probably_ a valid source that gets the driver files from Prolific, but I always download from the provider directly when I can. I believe that's the preference of the caskroom project as well. 

--

1. I understand that Mountain Lion isn't officially supported. That was more of a secondary change I offered because I was already making modifications. I haven't used Mountain Lion in years, either. 
2. This I totally get, and is the _most valid_ reason for rejection that I can see here. I don't really know why they've done what they did, or what they way around it is. 
3. I'm all for the web being blanketed in HTTPS. I feel like with the proliferation of providers and services like letsencrypt, there isn't really an excuse anymore. With that said, is lack of HTTPS on a download URL really the most critical reason for rejection? It's a _downloaded_ file with a known SHA hash. We're not transmitting PII such as names addresses or passwords, but a file that already exists in the public domain and space. Again, I'd love to see HTTPS on all things in all spaces, but it does not appear to be an option on the _official_ source for these drivers. I've always understood (at least in the main caskroom project) that HTTPS urls for downloads were preferred if available, but not required. 

I'm fine letting this be, I just wanted to offer some insight as to why I did what I did in case any of these topics come up again, in the scope of this project. Thanks for taking the time to review the PR and the time to read my (and supply your own) comments!

**Comment by vitorgalvao** [comment]
You make good points. Also, from the history of this cask in the main repo, I see it’s domain was actually the `.com.tw` in the beginning and it was changed later.

I’m not adamant on either accepting or refusing the change, so I’ll reopen with a few corrections and ping @commitay to see if there’s something I haven’t considered.

**Comment by vitorgalvao** [comment]
From the history of this cask in the main repo, I see it’s domain was actually the `.com.tw` in the beginning and it was changed later.

You make good points I’m not adamant on either accepting or refusing the change, so I’ve reopened with a few corrections and am pinging @commitay to see if there’s something I haven’t considered.

But so far I’m more for keeping the USA site so we keep HTTPS and `version` is simpler.

File: Casks/prolific-pl2303.rb [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
-  sha256 '14868e4a3c38904760d3445c37bbb5ca2f1024498547645147abbabfcb52eb87'
+  if MacOS.version <= :mountain_lion
+    version '1.5.1_20160309'
+    sha256 'b61561766d1bc1edd520a671a6dbe06fe81016eca0db76d37eeda2f481f53798'
+    url "http://www.prolific.com.tw/UserFiles/files/PL2303_MacOSX__v#{version.dots_to_underscores}.zip"
+    pkg "PL2303_MacOSX_v#{version}.pkg"
+  else
+    version '1.6.1_20160309'
+    sha256 '75ae5fa00c3862cfe7228a16dde06fb234a1e0b588bf7c4e29edde591a595a68'
+    url "http://www.prolific.com.tw/UserFiles/files/PL2303_MacOSX_#{version.dots_to_underscores}.zip"
```
Comment (vitorgalvao) [inline Casks/prolific-pl2303.rb]: Shasum computes to the same file, so there’s no point in keeping the 2017 version.

File: Casks/prolific-pl2303.rb [hunk 2]
```diff
@@ -1,12 +1,18 @@
 cask 'prolific-pl2303' do
-  version '1.6.1_20160309'
-  sha256 '14868e4a3c38904760d3445c37bbb5ca2f1024498547645147abbabfcb52eb87'
+  if MacOS.version <= :mountain_lion
+    version '1.5.1_20160309'
+    sha256 'b61561766d1bc1edd520a671a6dbe06fe81016eca0db76d37eeda2f481f53798'
+    url "http://www.prolific.com.tw/UserFiles/files/PL2303_MacOSX__v#{version.dots_to_underscores}.zip"
+    pkg "PL2303_MacOSX_v#{version}.pkg"
+  else
+    version '1.6.1_20160309'
```
Comment (vitorgalvao) [inline Casks/prolific-pl2303.rb]: Checksum computes to the same file, so there’s no point in keeping the 2017 version.

File: Casks/prolific-pl2303.rb [hunk 3]
```diff
@@ -24,8 +24,4 @@
                        '/var/db/receipts/*PL2303*.*',
                        '/var/db/receipts/*ProlificUSbSerial*.*',
                      ]
-
-  caveats do
-    kext
```
Comment (commitay) [inline Casks/prolific-pl2303.rb]: `kext` caveat is redundant with `uninstall kext:`

**Pull request merged**